openquake/hazardlib/gsim/germany_2018.py

```diff
@@ -0,0 +1,348 @@
+                       `DerrasEtAl2014RhypoMSD`,
+                       `DerrasEtAl2014RhypoHSD`,
+                       `BindiEtAl2017RhypoLSD,
+                       `BindiEtAl2017RhypoMSD,
+                       `BindiEtAl2017RhypoHSD
+"""
+import numpy as np
+from scipy.constants import g
+from openquake.hazardlib.imt import PGA, SA
+from openquake.hazardlib.gsim.akkar_2014 import AkkarEtAlRhyp2014
+from openquake.hazardlib.gsim.bindi_2014 import BindiEtAl2014Rhyp
+from openquake.hazardlib.gsim.cauzzi_2014 import CauzziEtAl2014
+from openquake.hazardlib.gsim.derras_2014 import DerrasEtAl2014
+from openquake.hazardlib.gsim.bindi_2017 import BindiEtAl2017Rhypo
+
+
+STRESS_DROP_ADJUST = {
+    "L": np.log(0.75),
+    "M": np.log(1.25),
+    "H": np.log(1.5)}
+
+
+def adjustment_fact(median, branch):
+    """
+    Applies stress drop adjustment factors from the STRESS_DROP_ADJ dictionary
+
+    :param numpy.ndarray median:
+        Median ground motions
+
+    :param str branch:
+        Key to indicate the branch
+    """
+    return median + STRESS_DROP_ADJUST[branch]
+
+
+class AkkarEtAlRhyp2014LSD(AkkarEtAlRhyp2014):
+    """
+    Akkar et al. (2014) "Rhypo" model with the "Low" stress drop adjustment
+    for Germany
+    """
+    SD_BRANCH = "L"
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        Returns the adjusted mean and standard deviation
+        """
+        mean, stddevs = super().get_mean_and_stddevs(sites, rup, dists, imt,
+                                                     stddev_types)
+        return mean + STRESS_DROP_ADJUST[self.SD_BRANCH], stddevs
+
+
+class AkkarEtAlRhyp2014MSD(AkkarEtAlRhyp2014LSD):
+    """
+    Akkar et al. (2014) "Rhypo" model with the "Middle" stress drop adjustment
+    for Germany
+    """
+    SD_BRANCH = "M"
+
+
+class AkkarEtAlRhyp2014HSD(AkkarEtAlRhyp2014LSD):
+    """
+    Akkar et al. (2014) "Rhypo" model with the "High" stress drop adjustment
+    for Germany
+    """
+    SD_BRANCH = "H"
+
+
+# Bindi et al 2014
+class BindiEtAl2014RhypLSD(BindiEtAl2014Rhyp):
+    """
+    Bindi et al. (2014) "Rhypo" model with the "Low" stress drop adjustment for
+    Germany
+    """
+    SD_BRANCH = "L"
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        Returns the adjusted mean and standard deviation
+        """
+        mean, stddevs = super().get_mean_and_stddevs(sites, rup, dists, imt,
+                                                     stddev_types)
+        return mean + STRESS_DROP_ADJUST[self.SD_BRANCH], stddevs
+
+
+class BindiEtAl2014RhypMSD(BindiEtAl2014RhypLSD):
+    """
+    Bindi et al. (2014) "Rhypo" model with the "Middle" stress drop adjustment
+    for Germany
+    """
+    SD_BRANCH = "M"
+
+
+class BindiEtAl2014RhypHSD(BindiEtAl2014RhypLSD):
+    """
+    Bindi et al. (2014) "Rhypo" model with the "High" stress drop adjustment
+    for Germany
+    """
+    SD_BRANCH = "H"
+
```

Comment:
In that case, given that the "L", "M" and "H" simply correspond to multiplicative factors 0.75, 1.25 and 1.5 respectively, I could simply do:
```python
class BindiEtAl2014Rhyp(...):
    def __init__(self, stress_drop_scaling=1.0):
        self.stress_drop_scaling = np.log(stress_drop_scaling)

    def get_mean_and_stddevs(...):
        mean, stddevs = super().get_mean_and_stddevs(...)
        return mean + self.stress_drop_scaling, stddevs
```
Thus the factors can be adjusted to whatever the user wishes.

Reply:
Indeed, that would be much better.

Reply:
Do the additional inputs to the GMPE always take the form of a string? i.e. I should do:
```python
class BindiEtAl2014Rhyp(...):
    def __init__(self, stress_drop_scaling="1.0"):
        self.stress_drop_scaling = np.log(float(stress_drop_scaling))
```
in order to avoid making changes in the GMPE logic tree reader?

Reply:
I don't remember, but I think there is a `literal_eval` there, so you can use any Python valid literal (including floats, integers, lists, tuples. dictionaries, None...). It not, I will add that. Just try.